include/aws/common/math.msvc.inl

```diff
@@ -44,21 +44,36 @@ AWS_STATIC_IMPL int aws_mul_u64_checked(uint64_t a, uint64_t b, uint64_t *r) {
  * Multiplies a * b. If the result overflows, returns 2^32 - 1.
  */
 AWS_STATIC_IMPL uint32_t aws_mul_u32_saturating(uint32_t a, uint32_t b) {
-    uint32_t out;
-    uint32_t ret_val = _mulx_u32(a, b, &out);
-    return (out == 0) ? ret_val : UINT32_MAX;
+    if (aws_cpu_has_feature(AWS_CPU_FEATURE_BMI2)) {
+        uint32_t out;
+        uint32_t ret_val = _mulx_u32(a, b, &out);
```

Comment:
is out here just whether or not overflow occured?

Reply:
It stores the high 32 bit from [here](https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mulx_u32&ig_expand=5244).

> Multiply unsigned 32-bit integers a and b, store the low 32-bits of the result in dst, and store the high 32-bits in hi

So, yeah, basically if it's non-zero, overflow happened.

Reply:
nit: but can we rename out to overflow or something?